code/src/components/Thoughts list/ThoughtsList.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { formatDistance } from 'date-fns';
+
+import './style.css';
+/* eslint no-underscore-dangle: 0 */
+const ThoughtsList = ({ loading, thoughtsList, setThoughtsList, onNewHearthSubmit }) => {
+  if (loading) {
+    return <h1>Loading in progres...</h1>
+  }
+  const onToughtCheckChange = (thought) => {
+    setThoughtsList(thoughtsList => thoughtsList.map(singleThought => {
+      if (singleThought._id === thought._id) {
+        return { ...singleThought, isChecked: !singleThought.isChecked };
+      }
+      return singleThought;
+    }));
+  }
+
+  return (
+    <section>
+      {thoughtsList.map(thought => (
```

Comment:
Im sure now that the reverse got out , you can finish the code. You have all that is needed so good job in grasping all the context, your code would go over the blue level. Just needing a very tiny polishment  for code completion . =)